### PR TITLE
Rely more on lru_cache rather than custom caching.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

All's in the title.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by jklymak** [comment]
@anntzer is this a soln to #9548?

**Comment by anntzer** [comment]
No, but I noticed the possibility of this improvement while reviewing your PR.

File: lib/matplotlib/font_manager.py [hunk 1]
```diff
@@ -1342,26 +1353,21 @@ def findfont(self, prop, fontext='ttf', directory=None,
...
-        result = _is_opentype_cff_font_cache.get(filename)
-        if result is None:
-            with open(filename, 'rb') as fd:
-                tag = fd.read(4)
-            result = (tag == b'OTTO')
-            _is_opentype_cff_font_cache[filename] = result
-        return result
-    return False
+        with open(filename, 'rb') as fd:
+            tag = fd.read(4)
```
Comment (Kojoley) [inline lib/matplotlib/font_manager.py]:
```python
            return fd.read(4) == b'OTTO'
```

File: lib/matplotlib/font_manager.py [hunk 2]
```diff
@@ -1268,6 +1269,20 @@ def findfont(self, prop, fontext='ttf', directory=None,
...
         """
+
+        # Pass the relevant rcParams (and the font manager, as `self`) to
+        # _findfont_cached so to prevent using a stale cache entry after an
+        # rcParam was changed.
+        return self._findfont_cached(
+            prop, fontext, directory, fallback_to_default, rebuild_if_missing,
+            tuple(tuple(rcParams[key]) for key in [
+                "font.serif", "font.sans-serif", "font.cursive",
+                "font.fantasy", "font.monospace"]))
```
Comment (Kojoley) [inline lib/matplotlib/font_manager.py]: This inlining...

File: lib/matplotlib/font_manager.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/font_manager.py]: fixed

File: lib/matplotlib/font_manager.py (hunk 2 above)
Comment (anntzer) [reply lib/matplotlib/font_manager.py]: deinlined :)

**Comment by tacaswell** [comment]
Thanks @anntzer !

Good to see you around reviewing again @Kojoley

**Pull request merged**